File: backend-intermediate-cs/OrderManagement/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OrderManagement.Models {
  [Table("product")]
  public class Product {
    [Key]
    [Column("product_id")]
    public int ProductId { get; set; }
    
    [Required]
    [Column("product_name")]
    public string ProductName { get; set; } = string.Empty;
    
    [Required]
    [Column("price")]
    [Range(0.01, double.MaxValue, ErrorMessage = "Price must be greater than 0")]
    public double Price { get; set; }
    
    [Required]
    [Column("category_id")]
    [Range(1, int.MaxValue, ErrorMessage = "Category ID must be greater than 0")]
    public int CategoryId { get; set; }
    
    // Navigation property
    public virtual Category? Category { get; set; }
  }
}

File: qa-fundamentals/OrderManagementJava/src/main/java/org/orders/model/Product.java
package org.orders.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "product")
@Getter
@Setter
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Integer productId;

    @NotBlank
    @Column(name = "product_name", nullable = false)
    private String productName;

    @NotNull
    @DecimalMin(value = "0.01", message = "Price must be greater than 0")
    @Column(name = "price", nullable = false)
    private Double price;

    @NotNull
    @Min(value = 1, message = "Category ID must be greater than 0")
    @Column(name = "category_id", nullable = false)
    private Integer categoryId;

    // Navigation property
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private Category category;
}
